python/hail/linalg/blockmatrix.py

```diff
@@ -387,21 +386,15 @@ def write_from_entry_expr(entry_expr, path, block_size=None):
         if not block_size:
             block_size = BlockMatrix.default_block_size()
 
-        source = entry_expr._indices.source
-        if not isinstance(source, MatrixTable):
-            raise ValueError("Expect an expression of 'MatrixTable', found {}".format(
-                "expression of '{}'".format(source.__class__) if source is not None else 'scalar expression'))
+        mt = matrix_table_source('write_from_entry_expr/entry_expr', entry_expr)
 
-        if entry_expr._indices != source._entry_indices:
-            from hail.expr.expressions import ExpressionException
-            raise ExpressionException("from_entry_expr: 'entry_expr' must be entry-indexed,"
-                                      " found indices {}".format(list(entry_expr._indices.axes)))
-
-        if entry_expr in source._fields_inverse:
-            source._jvds.writeBlockMatrix(path, source._fields_inverse[entry_expr], block_size)
+        if entry_expr in mt._fields_inverse:
```

Comment:
these two branches have different semantics. A row or column field will work in the 'else', because you can do `mt.select_entries(x = <row field>)`. I'm not sure which is correct (being strict or permissive) but it seems like this performance hack shouldn't have different behavior.

Reply:
mostly I don't really like the fields_inverse trick, though understand why it's done for now. I guess if this is marked with a fixme to delete ASAP, that's fine.

Reply:
Doesn't this check in `select_entries` prevent a row field from working?
```
            if not e._indices == self._entry_indices:
                # detect row or col fields here
                raise ExpressionException("method 'select_entries' parameter 'exprs' expects entry-indexed fields,"
                                          " found indices {}".format(list(e._indices.axes)))
```

Reply:
only as varargs, not with kwargs.